lib/teiserver/battle/balance/split_one_chevs.ex

```diff
@@ -0,0 +1,205 @@
+  %{rating: 7, member_id: 3, rank: 0}
+  ]
+
+  Returns %{teams:teams, logs:logs}
+  """
+  def assign_teams(member_list, number_of_teams) do
+    default_acc = %{teams: create_empty_teams(number_of_teams), logs: ["Begin split_one_chevs balance"]}
+
+    Enum.reduce(member_list, default_acc, fn x, acc ->
+      picking_team = get_picking_team(acc.teams)
+      update_picking_team = Map.merge(picking_team, %{members: [x | picking_team.members]})
+      username = Account.get_username_by_id(x.member_id)
+      new_log = "#{username} (Chev: #{x.rank+1}) picked for Team #{picking_team.team_id}"
+
+      %{
+        teams: [update_picking_team | get_non_picking_teams(acc.teams, picking_team)],
+        logs: acc.logs ++ [new_log]
+      }
+    end)
+  end
+
+  @spec create_empty_teams(any()) :: any()
+  def create_empty_teams(count) do
+    for i <- 1..count,
+        do: %{team_id: i, members: []}
+  end
+
+  @spec get_picking_team(any()) :: any()
+  def get_picking_team(teams) do
+    default_picking_team = Enum.at(teams, 0)
+
+    Enum.reduce(teams, default_picking_team, fn x, acc ->
+      # Team is picker if it has least members
+      if(length(x.members) < length(acc.members)) do
+        x
+      else
+        # Team is picker if it is tied for least and has lower team rating
+        if(
+          length(x.members) == length(acc.members) && get_team_rating(x) < get_team_rating(acc)
+        ) do
+          x
+        else
+          acc
+        end
+      end
+    end)
+  end
+
+  def get_non_picking_teams(teams, picking_team) do
+    Enum.filter(teams, fn x -> x.team_id != picking_team.team_id end)
+  end
+
+  def get_team_rating(team) do
+    Enum.reduce(team.members, 0, fn x, acc ->
+      acc + x.rating
+    end)
+  end
+
+  @doc """
+  teams=
+     [
+             %{
+               members: [
+                 %{rating: 17, rank: 0, member_id: 3},
+                 %{rating: 8, rank: 4, member_id: 100}
+               ],
+               team_id: 1
+             },
+             %{
+               members: [
+                 %{rating: 6, rank: 0, member_id: 2},
+                 %{rating: 5, rank: 0, member_id: 4}
+               ],
+               team_id: 2
+             }
+           ]
+  """
+  def standardise_result(teams, logs) do
+    team_groups = standardise_team_groups(teams)
+
+    %{
+      team_groups: team_groups,
+      team_players: standardise_team_players(teams),
+      logs: logs
+    }
+  end
+
+  def standardise_team_groups(raw_input) do
+    Map.new(raw_input, fn x -> {x.team_id, standardise_members(x.members)} end)
+  end
+
+  @doc """
+  members=
+  [
+                 %{rating: 6, rank: 0, member_id: 2},
+                 %{rating: 5, rank: 0, member_id: 4}
+               ]
+  output= [
+                 %{members: [2], count: 1, group_rating: 6, ratings: [6]},
+                 %{members: [4], count: 1, group_rating: 5, ratings: [5]}
+               ]
+  """
+  def standardise_members(members) do
```

Comment:
In general, prefer to use `defp` for functions that are purely internal to your module, makes easier to know what's safe to touch from what requires a bit more thought.